contracts/ERC721A.sol

```diff
@@ -684,26 +699,26 @@ contract ERC721A is IERC721A {
         uint256 prevOwnershipPacked = _packedOwnershipOf(tokenId);
 
         address from = address(uint160(prevOwnershipPacked));
-        address approvedAddress = _tokenApprovals[tokenId];
 
-        if (approvalCheck) {
-            bool isApprovedOrOwner = (_msgSenderERC721A() == from ||
-                isApprovedForAll(from, _msgSenderERC721A()) ||
-                approvedAddress == _msgSenderERC721A());
+        (uint256 approvedAddressSlot, address approvedAddress) = _getApprovedAddress(tokenId);
 
-            if (!isApprovedOrOwner) revert TransferCallerNotOwnerNorApproved();
-        }
+        if (approvalCheck)
+            if (!_isOwnerOrApproved(approvedAddress, from, _msgSenderERC721A()))
+                if (!isApprovedForAll(from, _msgSenderERC721A())) revert TransferCallerNotOwnerNorApproved();
```

Comment:
Can you combine this with the if-statement on 706? The formatting looks weird right now.

Reply:
This is intentional for performance.

Reply:
26

Reply:
Can you add a comment here then? I know people are going to ask about this otherwise.

Reply:
Ok, added